inpsyde/php-coding-standards
# Modernization

This brach if merged will:

- Modernize the codebase:
  - Move from PSR-2 or PSR-12
  - Use latest PHPCS
  - Use latest WPCS
  - Use latest VIP CS now in separate repo from WPCS

- Cleanup:
  - Remove custom and 3rd party sniffs that are duplicated
  - Remove `PHPCSAliases.php` that's no more necessary
  - Remove `arg` and `config` from `ruleset.xml` 

Improve:
 - Make customization of ElementNameMinimalLength sniff
 - Improve detection of hook callback

Overall, merging this PR will fix:

- #21 
- #25
- #26 (some outdated packages will be there because of PHPUnit 6 that we need due to PHP 7.0 support)
- #29
- #33 

So basically all open issues.

Before merging this has to be tested in the real world, that is, the updated repo as to be used to check a few WP codebases.

When merged we can release v0.15 which could be the 1.0-beta.
